src/protocol/sort/bit_permutation.rs

```diff
@@ -43,18 +52,21 @@ impl<'a, F: Field> BitPermutation<'a, F> {
     ///
     /// ## Errors
     /// It will propagate errors from multiplication protocol.
-    pub async fn execute(
-        &self,
-        ctx: ProtocolContext<'_, Replicated<F>, F>,
-    ) -> Result<Vec<Replicated<F>>, BoxError> {
-        let share_of_one = Replicated::one(ctx.role());
+    pub async fn execute(&self, ctx: ProtocolContext<'a, S, F>) -> Result<Vec<S>, BoxError>
+    where
+        ProtocolContext<'a, S, F>: SecureMul<F, Share = S>,
+    {
+        let share_of_one = S::one(
+            ctx.role(),
+            ctx.narrow(&ShareOfOne).prss().generate_replicated(RECORD_0),
+        );
 
         let mult_input = self
             .input
             .iter()
-            .map(move |x: &Replicated<F>| share_of_one - *x)
+            .map(move |x: &S| share_of_one - *x)
             .chain(self.input.iter().copied())
-            .scan(Replicated::<F>::new(F::ZERO, F::ZERO), |sum, x| {
+            .scan(S::default(), |sum, x| {
```

Comment:
Need to add a test to show it works for malicious. Let me know if it looks otherwise